Source/WebCore/inspector/NetworkResourcesData.cpp

```diff
@@ -256,10 +257,18 @@ void NetworkResourcesData::maybeDecodeDataToContent(const String& requestId)
     if (!resourceData->hasData())
         return;
 
-    m_contentSize += resourceData->decodeDataToContent();
-    size_t dataLength = resourceData->content().sizeInBytes();
-    if (dataLength > m_maximumSingleResourceContentSize)
-        m_contentSize -= resourceData->evictContent();
+    auto dataLength = resourceData->dataLength();
+    m_contentSize -= dataLength;
+
+    resourceData->decodeDataToContent();
```

Comment:
can you explain more why the code that was there before is not good enough?

additionally, with this change does `NetworkResourcesData::ResourceData::decodeDataToContent` need to return a value anymore?  if not, can we adjust that function to remove that unnecessary logic?